Test/comp/ComprehensionsNewSyntax.dfy

```diff
@@ -0,0 +1,284 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %dafny /noVerify /compile:4 /spillTargetCode:2 /compileTarget:cs "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /spillTargetCode:2 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /spillTargetCode:2 /compileTarget:go "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /spillTargetCode:2 /compileTarget:java "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+
+method Main() {
+  Quantifier();
+  MapComprehension();
+  OutParamsUnderLambdas();  // white-box testing
+  SetComprehension();
+  Enumerations();
+  EnumerationsMaybeNull();
+  TestImplicitTypeTests.Test();
+}
+
+predicate method Thirteen(x: int) { x == 13 }
+predicate method Even(y: int) { y % 2 == 1 }
+function method FourMore(x: int): int { x + 4 }
+
+method Quantifier() {
+  // Note, by making *two* assignments to "s" (instead of the obvious one), the direct translation
+  // into Java makes "s" NOT be "effectively final". This means that Java does not allow "s" to be
+  // used in the lambda expression into which the quantifier translates. Compilation into Java
+  // thus needs to capture the value of "s" into another variable, which these tests check on.
+  var s := [0, 1, 1, 2];
+  s := s + [3, 5, 8, 13];
+  print forall x <- s :: x < 20, " ";  // true
```

Comment:
I'm so happy to see the reduction in the duplication of the variable x! Great job!

Reply:
Oh, I didn't realize that this would be supported in `forall`/`exists`!  I thought it was just for `set`/`seq`/`multiset`

Reply:
Might as well! They all have the same underlying quantifier domain semantics, so I like the simplicity of aligning the syntax.

Reply:
Still worth checking a bit more broadly, maybe ^^

Reply:
I'm afraid I'm not following, checking what exactly? This was the intention in the RFC all along at least.

Reply:
It was unexpected to me, despite having read the proposals, so I'm guessing it might be unexpected to other people who provided feedback as well; that's it :)